src/Symfony/Component/ErrorHandler/Tests/ErrorHandlerTest.php

```diff
@@ -123,11 +123,57 @@ public function testNotice()
     }
 
     // dummy function to test trace in error handler.
-    private static function triggerNotice($that)
+    public static function triggerNotice($that)
     {
         $that->assertSame('', $foo.$foo.$bar);
     }
 
+    public function testFailureCall()
+    {
+        $this->expectException(\ErrorException::class);
+        $this->expectExceptionMessage('fopen(unknown.txt): failed to open stream: No such file or directory');
+
+        ErrorHandler::call('fopen', 'unknown.txt', 'r');
+    }
+
+    public function testCallRestoreErrorHandler()
+    {
+        $prev = set_error_handler('var_dump');
+        try {
+            ErrorHandler::call('fopen', 'unknown.txt', 'r');
+        } catch (\ErrorException $e) {
+            $prev = set_error_handler($prev);
```

Comment:
After we should restore the error handler twice

Reply:
Done.